src/Service/RegisteredMethodManager.php

```diff
@@ -104,6 +105,33 @@ public function registerForMember(Member $member, MethodInterface $method, $data
         return true;
     }
 
+    /**
+     * Determines if a method can be removed
+     *
+     * By default this is false if MFA is required and the method is the last on the Member (besides the backup method)
+     * but the funcation provides a hook point for extensibility e.g. if an site requires a particular method to be in
+     * use by a subset of members - admins must use U2F but normal users can use TOTP.
+     *
+     * @param Member $member
+     * @param MethodInterface $method
+     * @return bool
+     */
+    public function canRemoveMethod(Member $member, MethodInterface $method): bool
+    {
```

Comment:
I feel like this is better suited as an exception (maybe as `assertCanRemoveMethod`). The UI should prevent getting here and an exception can provide more context (like the fact it should be a `403`). Keen to hear thoughts.

Reply:
I feel like this would be unnecessary. The method could be used in other contexts than mid-removal, such as whether or not to show an action to remove - not just in the middle of a removal request. In the case that it would throw and exception would introduce a try-catch statement, and/or having to deal with an HTTP error for what should essentially be a boolean.